src/components/tabs/ScatterPlotTab.tsx

```diff
@@ -411,78 +417,105 @@ export default memo(function ScatterPlotTab(props: ScatterPlotTabProps): ReactEl
     yData: DataArray,
     objectIds: number[],
     trackIds: number[],
-    markerConfig: Partial<PlotMarker> = {},
+    markerConfig: Partial<PlotMarker> & { outliers?: Partial<PlotMarker>; outOfRange?: Partial<PlotMarker> } = {},
     overrideColor?: Color
   ): Partial<PlotData>[] => {
     if (selectedFeatureName === null || dataset === null || !xAxisFeatureName || !yAxisFeatureName) {
       return [];
     }
-
     const featureData = dataset.getFeatureData(selectedFeatureName);
+    if (!featureData) {
+      return [];
+    }
 
-    const traceDataBuckets: TraceData[] = [];
-    if (!featureData || markerConfig.color || overrideColor) {
-      // Do no coloring! Keep all points in the same bucket.
-      traceDataBuckets.push({
-        x: Array.from(xData),
-        y: Array.from(yData),
-        objectIds: objectIds,
-        trackIds: trackIds,
-        color: overrideColor || new Color(),
-      });
+    // Generate colors
+    const categories = dataset.getFeatureCategories(selectedFeatureName);
+    const isCategorical = categories !== null;
+    const usingOverrideColor = markerConfig.color || overrideColor;
+    overrideColor = overrideColor || new Color(markerConfig.color as ColorRepresentation);
+
+    let colors: Color[];
+    if (usingOverrideColor) {
+      // Use a single color bucket
+      colors = [overrideColor];
+    } else if (isCategorical) {
+      colors = categoricalPalette.slice(0, categories.length);
     } else {
```

Comment:
This caused a bug where hidden outliers/out-of-range values would be mistakenly revealed on the plot, since the hiding/revealing of values is handled by what used to be the `else` branch. I merged everything into one code pathway here.